tests/unit/lms/resources/_js_config/__init___test.py

```diff
@@ -567,57 +567,67 @@ def config(self, config, js_config):
 class TestEnableOAuth2RedirectErrorMode:
     def test_scope_error(self, js_config):
         js_config.enable_oauth2_redirect_error_mode(
-            auth_url=None,
+            auth_route="auth_route",
             error_details="Technical error",
             is_scope_invalid=True,
             canvas_scopes=["scope_a", "scope_b"],
         )
 
         config = js_config.asdict()
-
         assert config["mode"] == "oauth2-redirect-error"
         assert config["oAuth2RedirectError"] == {
-            "authUrl": None,
+            "authUrl": "http://example.com/auth?authorization=xyz123",
             "invalidScope": True,
             "errorDetails": "Technical error",
             "canvas_scopes": ["scope_a", "scope_b"],
         }
 
     def test_other_error(self, js_config):
-        auth_url = "https://lms.hypothes.is/auth/url"
         js_config.enable_oauth2_redirect_error_mode(
-            auth_url=auth_url, error_details="Some error"
+            auth_route="auth_route", error_details="Some error"
         )
 
         config = js_config.asdict()
-
-        assert config["mode"] == "oauth2-redirect-error"
         assert config["oAuth2RedirectError"] == {
-            "authUrl": auth_url,
+            "authUrl": "http://example.com/auth?authorization=xyz123",
             "invalidScope": False,
             "errorDetails": "Some error",
             "canvas_scopes": [],
         }
 
-    @pytest.fixture
-    def js_config(self, context, pyramid_request):
-        return JSConfig(context, pyramid_request)
+    @pytest.mark.usefixtures("no_user")
+    def test_auth_url(self, js_config):
+        js_config.enable_oauth2_redirect_error_mode(
+            auth_route="auth_route", error_details="Some error"
+        )
+
+        config = js_config.asdict()
+        assert config["oAuth2RedirectError"]["authUrl"] is None
 
     @pytest.fixture
-    def pyramid_request(self, pyramid_request):
+    def no_user(self, pyramid_request):
```

Comment:
```suggestion
    def with_no_user(self, pyramid_request):
```
Seems to be a pattern we have elsewhere?

Reply:
:+1: Done